arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,491 @@
+Run `yarn hardhat compile` to compile the default `contracts`. You may be prompted to install additional dependencies - follow those instructions until this command runs successfully. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/` directory. This directory contains the compiled smart contract.
+
+Open `scripts/deploy.js` and replace its contents with the following:
+
+```javascript title="scripts/deploy.js"
+const hre = require("hardhat");
+
+async function main() {
+
+  const VendingMachineFactory = await hre.ethers.getContractFactory("VendingMachine");
+  const vendingMachine = await VendingMachineFactory.deploy();
+  await vendingMachine.deployed();
+
+  console.log(
+    `Cupcake vending machine deployed to ${vendingMachine.address}`
+  );
+}
+
+main().catch((error) => {
+  console.error(error);
+  process.exit(1);
+});
+```
+
+We'll use this to deploy our smart contract in a moment. Next, delete `contracts/Lock.sol` and replace it with `contracts/VendingMachine.sol`, the smarter alternative to our Javascript implementation:
+
+```solidity title="VendingMachine.sol"
+pragma solidity ^0.8.9;
+
+// Rule 2: The vending machine's rules can't be changed by anyone.
+contract VendingMachine {
+    // state variables = internal memory of the vending machine
+    mapping(address => uint) private _cupcakeBalances;
+    mapping(address => uint) private _cupcakeDistributionTimes;
+
+    function giveCupcakeTo(address userAddress) public returns (bool) {
+        // this code is unnecessary, but we're keeping it here so you can compare it to the JS implementation
+        if (_cupcakeDistributionTimes[userAddress] == 0) {
+            _cupcakeBalances[userAddress] = 0;
+            _cupcakeDistributionTimes[userAddress] = 0;
+        }
+
+        // Rule 1: The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+        uint fiveSecondsFromLastDistribution = _cupcakeDistributionTimes[userAddress] + 5 seconds;
+        bool userCanReceiveCupcake = fiveSecondsFromLastDistribution <= block.timestamp;
+        if (userCanReceiveCupcake) {
+            _cupcakeBalances[userAddress]++;
+            _cupcakeDistributionTimes[userAddress] = block.timestamp;
+            return true;
+        } else {
+            revert("HTTP 429: Too Many Cupcakes (you must wait at least 5 seconds between cupcakes)");
+        }
+    }
+
+    // Getter function for the cupcake balance of a user
+    function getCupcakeBalanceFor(address userAddress) public view returns (uint) {
+        return _cupcakeBalances[userAddress];
+    }
+}
+```
+
+Note that this smart contract is written in Solidity, a language that compiles to EVM bytecode. This means that it can be deployed to any Ethereum-compatible blockchain, including Ethereum mainnet, <a data-quicklook-from='arbitrum-one'>Arbitrum One</a>, and <a data-quicklook-from='arbitrum-nova'>Arbitrum Nova</a>.
```

Comment:
Should we add some reference for "EVM bytecode" here?
Maybe a glossary term or a link that explains what it is.

Reply:
TODO (eventually): A new concept doc that ELI5 -> ELI30s this thing in the context of Arbitrum (and Stylus)

(I'll find something else to link to in the meantime)